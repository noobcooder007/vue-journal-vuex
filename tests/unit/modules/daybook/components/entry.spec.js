import { shallowMount } from '@vue/test-utils'
import Entry from '../../../../../src/modules/daybook/components/Entry'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Puerbas en Entry Component', () => {
    const mockRouter = {
        push: jest.fn()
    }
    const entry = { ...journalState.entries[0]}
    const wrapper = shallowMount(Entry, {
        props: {
            entry
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })
    test('Debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Debe de redireccionar al hacer click en el entry-container', () => {
        wrapper.find('.entry-container').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: entry.id } })
    })
    test('Pruebas en las propiedades computadas', () => {
        expect(wrapper.vm.day).toBe(17)
        expect(wrapper.vm.month).toBe('Agosto')
        expect(wrapper.vm.yearDay).toBe('2021, Martes')
    })
})