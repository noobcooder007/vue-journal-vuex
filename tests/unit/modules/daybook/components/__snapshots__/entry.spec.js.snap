// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puerbas en Entry Component Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">17</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-light">2021, Martes</span></div>
  <div class="entry-description">Este es un nuevo d√≠a con Vue.js</div>
</div>
`;
